lib/user.rb

```diff
@@ -0,0 +1,22 @@
+class User
+  attr_reader :name, :jokes
+  def initialize(name)
+    @name = name
+    @jokes = []
+  end
+
+  def learn(joke)
+    @jokes << (joke)
+  end
+
+  def tell(friend, joke)
+    friend.learn(joke)
+  end
+
+  def joke_by_id(id)
+    @jokes.find_all do |ident|
```

Comment:
One other small thing-- `ident` isn't a particularly good variable name in this case.  Since you're iterating over an array of joke objects, and the block variable represents each individual element, each individual element would be a `joke`, so that would probably be a better variable name here.